services/src/main/java/org/keycloak/services/resources/admin/RealmsAdminResource.java

```diff
@@ -93,14 +94,26 @@ public RealmsAdminResource(AdminAuth auth, TokenManager tokenManager) {
     @GET
     @NoCache
     @Produces(MediaType.APPLICATION_JSON)
-    public Stream<RealmRepresentation> getRealms() {
+    public Stream<RealmRepresentation> getRealms(@QueryParam("basic") String basic) {
```

Comment:
This call produces an array of hashes like `[{id: ..., realm: ...}, ...{id: ..., realm: ...}]`. This contains lot of garbage and can be optimized to only return array of pairs like `[["id1", "realm1"], ["id2", "realm2], ...]`
Could there be an endpoint "/list" that the following implementation:

```java
    @GET
    public Stream<String[]> listRealms() {
        return session.realms().getRealmsStream()
            map(r -> return new String[] {r.getId(), r.getName});
    }
```
This would remove all the logic linked to serialization of RealmRepresentation which would in turn lead to a more performant behaviour.

Reply:
Wouldn't that mean changing the console too? If so, I'm not sure if the effort justifies the performance gain.

I would leave it as is, please. Later we can see how to improve it even more.

Reply:
You're right, that could mean changing the console as well. Could the parameter have a more descriptive name, like `idNameOnly`?

Reply:
Basic can mean more than id and name. Ideally, we should be following the rest guidelines so that we can have better control over the response. But that is another work. I would just leave as is.

Reply:
For something else, there is already used `briefRepresentation`. Please keep parameter names consistent across code base

Reply:
I agree with renaming the parameter to `briefRepresentation` we are already using it in many places. One more question, should we default to `true`? I am not sure about all places where this endpoint is used, but I can't imagine any usecase where it is necessary to obtain all realms in one place, so true as the default value makes sense to me. WDYT?

Reply:
I'll change the name. A big one though :) But if we are using at other places, so let's keep consistency. Hope we can sort this out better when we review our APIs based on the guidelines.

In regards to default to true, that could break backward compatibility. I would leave it as is.